test/Bitmap/IndexedBmpWriter.test.cpp

```diff
@@ -10,12 +11,11 @@ TEST(BitmapFile, InvalidBitCountThrows)
 {
 	const std::vector<Color> palette(8);
 	const std::vector<uint8_t> pixels(4, 0);
-	const std::string filename = "Sprite/data/MonochromeTest.bmp";
+	Stream::DynamicMemoryWriter writer;
 
-	EXPECT_THROW(BitmapFile::CreateIndexed(3, 1, 1, palette, pixels).WriteIndexed(filename), std::runtime_error);
-	EXPECT_THROW(BitmapFile::CreateIndexed(32, 1, 1, palette, pixels).WriteIndexed(filename), std::runtime_error);
-
-	XFile::DeletePath(filename);
+	EXPECT_THROW(BitmapFile::CreateIndexed(3, 1, 1, palette, pixels).WriteIndexed(writer), std::runtime_error);
+	writer.Clear();
+	EXPECT_THROW(BitmapFile::CreateIndexed(32, 1, 1, palette, pixels).WriteIndexed(writer), std::runtime_error);
```

Comment:
Technically, I don't believe it's essential to call `Clear` between these two expectations since neither depends on anything written or not written to the stream. Though perhaps this is getting at a deeper problem.

Considering the two expectations are independent of each other, the code should ideally be written such that failures or side effects from one expectation can't inadvertently affect the later expectation. Calling `Clear` is an attempt to manage side effects to shared mutable state. Instead, the expectations should be written to avoid sharing mutable state.

If different errors are being tested, these checks could potentially be two separate unit tests. Seeing as how they are strongly related though, you might put them both in the same unit test, but use separate writers for each. If you want to re-use variable names, then wrap each check in a scope block:
```cpp
{
    Stream::DynamicMemoryWriter writer;
    EXPECT_THROW(BitmapFile::CreateIndexed(3, 1, 1, palette, pixels).WriteIndexed(writer), std::runtime_error);
}
{
    Stream::DynamicMemoryWriter writer;
    EXPECT_THROW(BitmapFile::CreateIndexed(32, 1, 1, palette, pixels).WriteIndexed(writer), std::runtime_error);
}
```